src/generate.ts

```diff
@@ -209,9 +226,9 @@ export function createGeneratePlugin({
         const dtsId = filename_ts_to_dts(resolution.id)
         if (inputAliasMap.has(dtsId)) return resolution
 
-        for (const [name, entry] of Object.entries(inputOption)) {
-          if (entry === id) {
-            inputAliasMap.set(dtsId, `${name}.d.ts`)
+        for (const [name, inputId] of Object.entries(inputOption)) {
+          if (inputId === id) {
+            inputAliasMap.set(id, name)
             break
           }
```

Comment:
I don't understand the propose of this chunk of code, and it's not been covered in the test cases I added. It would be better if we can add a test case to cover it, or remove it if it's unnecessary.

Reply:
Actually, I’m also a bit confused about this. @GoodbyeNJN, could you please provide a use case?

Reply:
```ts
import { defineConfig } from "rolldown";

export default defineConfig({
    input: {
        index: "./src/eslint/index.ts",
        prettier: "./src/prettier/index.ts",
    },
    output: {
        dir: "./dist",
        format: "esm",
    },
    plugins: [dts()],
});
```

The `inputId === id` is for matching the corresponding property name from the object `
{ index: "./src/eslint/index.ts", prettier: "./src/prettier/index.ts" }`.

The refactored code already covers the scenario I mentioned, and I think it's safe to remove that part of the code.